django/db/models/sql/query.py

```diff
@@ -57,6 +57,15 @@ def get_children_from_q(q):
             yield child
 
 
+def get_col_from_expression(expression):
+    # get Col from expression
+    targets = expression.get_source_expressions()
+    if targets and isinstance(targets[0], Col):
+        return targets[0]
+    else:
+        return None
```

Comment:
I don't think that it will work in all cases, you could use `_get_cols()` but it doesn't give you much. We need to know whether an expression is nullable or not, so we cannot take only columns into account. Some expressions are non-nullable even if all  the columns passed are.

Simon proposed to add a `nullable` to the `BaseExpression` (see https://github.com/django/django/pull/11600#discussion_r308211556), it would help solving multiple issues (e.g. ticket-25946, ticket-29291) and fits perfectly with this patch. Can you prepare a patch in advance with a new `nullable` property?

Reply:
@felixxm What you mean is to fix this bug after proceeding with inserting nullable property first?

Reply:
@felixxm Okay. I will try to add nullable to the BaseExpression.

Reply:
> @felixxm What you mean is to fix this bug after proceeding with inserting `nullable` property first?

Yes.

Reply:
@felixxm Currently i'm trying to add a `nullable` to `BaseExpression`.

I think there are two cases in this situation.

1) Expression which depends on the `column`.

ex) ExtractYear('created_at') -> depends on 'create_at' field is nullable or not.


2) Expression which doesn't depends on 'column'.

ex) Subquery -> it's always can be null.

I think I have to divide two cases. how about your opinion?

Reply:
I saw your comment in the previous [PR](https://github.com/django/django/pull/11600#discussion_r371791804).
```
This flag should be set properly for all expressions, e.g. in `Col` should contain the same logic as in is_nullable(), many text functions are not nullable on Oracle because it interprets empty strings as NULL's, e.g. MD5, Concat() is not nullable, etc.
```
Do I need to check all of the expressions? and if we need to check column, we have to add is_nullable() function, right?